experimenter/experimenter/experiments/api/v5/serializers.py

```diff
@@ -2102,6 +2102,41 @@ def _validate_desktop_pref_rollouts(self, data):
 
         return data
 
+    def _validate_desktop_pref_flips(self, data):
+        if not any(
+            fc.slug == NimbusConstants.DESKTOP_PREFFLIPS_SLUG
+            for fc in data.get("feature_configs", [])
+        ):
+            return data
+
+        min_version = NimbusExperiment.Version.parse(data["firefox_min_version"])
+        channel = data.get("channel")
+
+        if not channel:
+            raise serializers.ValidationError(
+                {
+                    "channel": NimbusConstants.ERROR_DESKTOP_PREFFLIPS_CHANNEL_REQUIRED,
+                }
+            )
+
+        # We have already validated that the feature is supported by the current
+        # version. However, we don't keep track of per-channel manifests. The
+        # prefFlips feature landed in 128 Nightly as a placeholder and was only
+        # implemented in 129 Nightly+, but it was uplifted to 128 ESR.
+        if channel == NimbusExperiment.Channel.ESR:
+            return data
```

Comment:
Like we allow any version on channel ESR?

Reply:
If we're hiting this point, we've already validated that the feature is available for this version. I'll add a comment.

Reply:
Oh. there already is a comment there.